packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceAddUser.tsx

```diff
@@ -215,7 +251,34 @@ export const ShareSpaceAddUser: FC<ShareSpaceAddUserProps> = ({
                 onChange={setUsersSelected}
                 data={data}
                 itemComponent={UserItemComponent}
-                filter={filterByNameOrEmail}
```

Comment:
No need to perform search by email here since the backend now takes care of all that

New filter function just let's all users go (ignores filtering), and matches the groups